Remove optional dependency flag for log4j-api

Gordiychuk opened the pull request:
Embedded version of elasticsearch and transport client can't work without
dependency on log4j-api, because they fail during initialize logging.

```
Caused by: java.lang.NoClassDefFoundError: org/apache/logging/log4j/Logger
	at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:105)
	at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:72)
	at org.elasticsearch.common.component.AbstractComponent.<init>(AbstractComponent.java:37)
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:98)
	at org.elasticsearch.client.transport.TransportClient.newPluginService(TransportClient.java:94)
	at org.elasticsearch.client.transport.TransportClient.buildTemplate(TransportClient.java:119)
	at org.elasticsearch.client.transport.TransportClient.<init>(TransportClient.java:247)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:125)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:111)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:101)
	... 99 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.Logger
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 120 common frames omitted
```

Now for start use Transport client or embedded version of elasticsearch
need add only one dependency

Closes #22671

karmi commented:
Hi @Gordiychuk, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/elasticsearch/pull/22674.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

clintongormley commented:
@rjernst please could you comment?

Gordiychuk committed:
Remove optional dependency flag for log4j-api

Embedded version of elasticsearch and transport client can't work without
dependency on log4j-api, because they fail during initialize logging.

```
Caused by: java.lang.NoClassDefFoundError: org/apache/logging/log4j/Logger
	at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:105)
	at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:72)
	at org.elasticsearch.common.component.AbstractComponent.<init>(AbstractComponent.java:37)
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:98)
	at org.elasticsearch.client.transport.TransportClient.newPluginService(TransportClient.java:94)
	at org.elasticsearch.client.transport.TransportClient.buildTemplate(TransportClient.java:119)
	at org.elasticsearch.client.transport.TransportClient.<init>(TransportClient.java:247)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:125)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:111)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:101)
	... 99 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.Logger
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 120 common frames omitted
```

Now for start use Transport client or embedded version of elasticsearch
need add only one dependency

Closes #22671

jasontedor commented:
As I [mentioned](https://github.com/elastic/elasticsearch/issues/22671#issuecomment-273460226) on #22671, I don't think we need to do this. I am not in the position to investigate this right now, but I will soon. Either way, I don't think we will pull this in.
